components/invitation_modal/invitation_modal_guest_step/invitation_modal_guests_step.tsx

```diff
@@ -22,30 +23,38 @@ import './invitation_modal_guests_step.scss';
 import {t} from 'utils/i18n.jsx';
 import {localizeMessage} from 'utils/utils.jsx';
 
-class InvitationModalGuestsStep extends React.PureComponent {
-    static propTypes = {
-        teamName: PropTypes.string.isRequired,
-        myInvitableChannels: PropTypes.array.isRequired,
-        currentTeamId: PropTypes.string.isRequired,
-        searchProfiles: PropTypes.func.isRequired,
-        searchChannels: PropTypes.func.isRequired,
-        defaultChannels: PropTypes.array,
-        defaultMessage: PropTypes.string,
-        onEdit: PropTypes.func.isRequired,
-        onSubmit: PropTypes.func.isRequired,
-        emailInvitationsEnabled: PropTypes.bool.isRequired,
-        userLimit: PropTypes.string.isRequired,
-        userIsAdmin: PropTypes.bool.isRequired,
-        isCloud: PropTypes.bool.isRequired,
-        subscriptionStats: PropTypes.object,
-        actions: PropTypes.shape({
-            getSubscriptionStats: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+import {PropsFromRedux} from './index';
+
+type OwnProps = {
+    teamName: string;
+    myInvitableChannels: Channel[];
+    currentTeamId: string;
+    searchProfiles: (term: string, options?: Record<string, any>) => Promise<{ data: UserProfile[] }>;
+    searchChannels: (teamId: string, term: string) => Promise<Channel[]>;
+    defaultChannels?: Channel[];
+    defaultMessage?: string;
+    onEdit: (hasChanges: boolean) => void;
+    onSubmit: (users: string[], emails: string[], channels: Channel[], message: string, extraUserText: string, extraChannelText: string) => Promise<void>;
+    emailInvitationsEnabled: boolean;
+}
+
+type Props = OwnProps & PropsFromRedux
+
+type State = {
+    customMessageOpen: boolean;
+    customMessage: string;
+    usersAndEmails: string[];
+    channels: Channel[];
+    usersInputValue: string;
+    channelsInputValue: string;
+    termWithoutResults?: string | null;
```

Comment:
Any way we can avoid using `null` here to simplify things?

Reply:
@devinbinnie it is from official documentation, https://react-redux.js.org/using-react-redux/usage-with-typescript#inferring-the-connected-props-automatically
But anyways i dont want to bring in new coding styles all of a sudden, let me know if i should manually write the types.

Reply:
> Any way we can avoid using `null` here to simplify things?

I added null because in the code it was being set to null, but i changed it to empty string instead and fixed the type